#!/bin/bash

#mkdir -p "output/tables/qc/tin.py"


#export cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam

export sid="AAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAP2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAS1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAS2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAT1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAT2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAU1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAU2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAV1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAV2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAW1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAW2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAX1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAX2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAY1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AAY2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ABA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ABA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ACA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ACA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ADA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ADA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AFA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AFA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AHA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AHA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AIA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AIA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AKA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AKA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ALA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ALA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AMA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AMA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AOA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AOA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AQA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AQA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="AZH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAO1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAO2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAR1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAR2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAS1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAS2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAT1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAT2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAU1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAU2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAW1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAW2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAX1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAX2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAY1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="BAY2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAO1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAO1-replicate"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAO2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAV1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CAV2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBP2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBQ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBQ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBR1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBR2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBS1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBS2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBT1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBT2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBV1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CBV2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCW1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCW2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCZ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CCZ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="CDH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="DAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAO1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAO2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAP2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAQ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAQ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAT1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAT2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAU1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAU2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAV1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAV2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAW1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAW2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAY1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAY2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAZ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EAZ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBO1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBO2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBR1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBR2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBU1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBU2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBV1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBV2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBW1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBW2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBX1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBX2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBY1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="EBY2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="ECN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAB2-replicate"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAP2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAQ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="FAQ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAO1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAO2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAP1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAP2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAQ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAQ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAR1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAR2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAS1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAS2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="GAS2-replicate"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="HAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAE2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAF1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAF2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAG1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAG2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAH1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAH2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAI1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAI2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAJ1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAJ2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAK1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAK2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAL1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAL2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAM1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAM2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAN1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="JAN2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAA1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAA2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAB1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAB2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAC1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAC2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAD1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAD2"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
export sid="KAE1"; cd "/home/yhoogstrate/projects/gsam"; mkdir -p "output/tables/qc/tin.py/"$sid ; cd "output/tables/qc/tin.py/"$sid ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam" $sid.bam ; ln -s "/home/yhoogstrate/mnt/neuro-genomic-ro/gsam/RNA/alignments/"$sid"/Aligned.sortedByCoord.out.markdup.bam.bai" $sid.bam.bai ; nice tin.py -r "/mnt/data/ccbc_environment/general/annotation/hg38/GENCODE/gencode.v30.annotation.bed" -s -i $sid.bam
