#!/bin/bash

mkdir -p output/tables/rsem


#export sid="AAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAP2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAS1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAS2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAT1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAT2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAU1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAU2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAV1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAV2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAW1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAW2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAX1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAX2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAY1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AAY2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ABA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ABA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ACA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ACA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ADA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ADA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AFA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AFA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AHA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AHA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AIA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AIA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AKA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AKA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ALA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ALA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AMA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AMA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AOA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AOA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AQA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AQA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="AZH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAO1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAO2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAR1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAR2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAS1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAS2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAT1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAT2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAU1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAU2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAW1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAW2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAX1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAX2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAY1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="BAY2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAO1-replicate"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAO1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAO2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAV1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CAV2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBP2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBQ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBQ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBR1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBR2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBS1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBS2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBT1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBT2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBV1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CBV2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCW1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCW2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCZ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CCZ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="CDH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="DAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAO1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAO2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAP2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAQ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAQ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAT1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAT2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAU1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAU2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAV1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAV2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAW1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAW2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAY1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAY2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAZ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EAZ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBO1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBO2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBR1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBR2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBU1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBU2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBV1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBV2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBW1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBW2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBX1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBX2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBY1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="EBY2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="ECN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAB2-replicate"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAP2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAQ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="FAQ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAO1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAO2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAP1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAP2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAQ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAQ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAR1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAR2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAS1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAS2-replicate"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="GAS2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="HAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAE2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAF1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAF2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAG1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAG2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAH1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAH2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAI1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAI2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAJ1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAJ2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAK1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAK2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAL1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAL2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAM1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAM2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAN1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="JAN2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAA1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAA2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAB1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAB2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAC1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAC2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAD1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAD2"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 
export sid="KAE1"; nice rsem-calculate-expression --bam --paired-end --forward-prob 0 --alignments --no-bam-output -p 8 "data/RNA/alignments/"$sid"/Aligned.toTranscriptome.out.bam" /mnt/data/ccbc_environment/project/gsam-neurology/ref/rsem-hg19/gencode.v31 "output/tables/rsem/"$sid 

